crates/sui-indexer/src/store/pg_indexer_store_v2.rs

```diff
@@ -771,6 +826,41 @@ impl IndexerStoreV2 for PgIndexerStoreV2 {
         Ok(())
     }
 
+    async fn persist_object_history(
+        &self,
+        object_changes: Vec<TransactionObjectChangesToCommit>,
+    ) -> Result<(), IndexerError> {
+        if object_changes.is_empty() {
+            return Ok(());
+        }
+        let objects = make_final_list_of_objects_to_commit(object_changes);
+        let guard = self
+            .metrics
+            .checkpoint_db_commit_latency_objects_history
+            .start_timer();
+
+        let len = objects.len();
+        let chunks = chunk!(objects, self.parallel_objects_chunk_size);
+        let futures = chunks
+            .into_iter()
+            .map(|c| self.spawn_blocking_task(move |this| this.persist_objects_history_chunk(c)))
+            .collect::<Vec<_>>();
+
+        futures::future::join_all(futures)
+            .await
+            .into_iter()
```

Comment:
I am just curious because I am not familiar with indexer code. How does the failure of one chunk is handled during retry or in other words how do we ensure we do not re-insert the same objects again when we retry on one chunk failure

Reply:
great q, if it's DB commit failure there is retry in `persist_objects_history_chunk`, it's on the chunk level and each chunk has different data to commit;
on the other hand, if the binary crashed and then resumed, indexer will read last_cp as the water mark and start ingestion, which might insert data that have been committed, so we do `on_conflict_do_nothing` to make it idempotent and thus would not have duplicate rows in DB.